logger/drain/factory.go

```diff
@@ -15,19 +14,14 @@ func NewDrain(drainURL string) (LogDrain, error) {
 		// nil means no drain-- which is valid
 		return nil, nil
 	}
-	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") {
-		drain, err := udp.NewDrain(drainURL)
-		if err != nil {
-			return nil, err
-		}
-		return drain, nil
-	}
-	if strings.HasPrefix(drainURL, "tcp://") {
-		drain, err := tcp.NewDrain(drainURL)
+	// Any of these three can use the same drain implementation
+	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") || strings.HasPrefix(drainURL, "tcp://") {
```

Comment:
Good question.  Skip a few lines down and there's a TODO with a partial explanation.  I expect other drains to be implemented soon-- either by us or the community.  I think those will differ more significantly from the one drain we have now.  I want the factory to be able to instantiate the correct sort of drain for whatever URL it's given.  It's only right now in the interim where there's only one drain implementation that this doesn't make so much sense, but it will soon.